### ISSUE TITLE
Introduce some backoff in the starvation warnings

### ISSUE THREAD

**Issue by djspiewak** [typelevel/cats-effect#3443]
When it triggers it tends to spam for a while. I'm not exactly certain how helpful that is. It definitely makes it hard to ignore, but it's also mostly noise. What we could do is start dynamically increasing the interval whenever it starts firing, up to some bound.

**Comment by aartigao** [comment]
If you were to touch this, would it be too much to ask to provide a configurable hook to allow printing the message in the user defined logging system? 🙏🏽 

Our company policy for logs uses JSON format for both sysout/syserr and those warnings break it. We can disable them, but I really would like to have them because they are helpful.

**Comment by armanbilge** [comment]
> would it be too much to ask to provide a configurable hook to allow printing the message in the user defined logging system

@aartigao Not at all! I suggested a similar idea in https://github.com/typelevel/cats-effect/issues/3308#issuecomment-1339712130 which had additional interest, and gave some suggestions how it may be implemented. It probably deserves its own issue, if you don't mind opening one, and would be an excellent first-time contribution :)
